airbyte-integrations/connectors/source-microsoft-sharepoint/source_microsoft_sharepoint/stream_reader.py

```diff
@@ -92,14 +106,65 @@ def config(self, value: SourceMicrosoftSharePointSpec):
         assert isinstance(value, SourceMicrosoftSharePointSpec)
         self._config = value
 
+    def _get_shared_drive_object(self, drive_id: str, object_id: str, path: str) -> List[Tuple[str, str, datetime]]:
+        """
+        Retrieves a list of all nested files under the specified object.
+
+        Args:
+            drive_id: The ID of the drive containing the object.
+            object_id: The ID of the object to start the search from.
+
+        Returns:
+            A list of tuples containing file information (name, download URL, and last modified datetime).
+
+        Raises:
+            RuntimeError: If an error occurs during the request.
+        """
+
+        access_token = self.get_access_token()
+        headers = {"Authorization": f"Bearer {access_token}"}
+        base_url = f"https://graph.microsoft.com/v1.0/drives/{drive_id}"
+
+        def get_files(url: str, path: str) -> List[Tuple[str, str, datetime]]:
+            response = requests.get(url, headers=headers)
+            if response.status_code != 200:
+                raise RuntimeError(f"Error retrieving shared files: {response.status_code}")
+
+            data = response.json()
+            for child in data.get("value", []):
+                new_path = path + "/" + child["name"]
+                if child.get("file"):  # Object is a file
+                    last_modified = datetime.strptime(child["lastModifiedDateTime"], "%Y-%m-%dT%H:%M:%SZ")
+                    yield (new_path, child["@microsoft.graph.downloadUrl"], last_modified)
+                else:  # Object is a folder, retrieve children
+                    child_url = f"{base_url}/items/{child['id']}/children"  # Use item endpoint for nested objects
+                    yield from get_files(child_url, new_path)
+            yield from []
+
+        # Initial request to item endpoint
+        item_url = f"{base_url}/items/{object_id}"
+        item_response = requests.get(item_url, headers=headers)
+        if item_response.status_code != 200:
+            raise RuntimeError(f"Error retrieving shared object: {item_response.status_code}")
```

Comment:
We need specified more user friendly error message with more details.

Reply:
Fixed